rosflight_sim/launch/multirotor_sim_io_joy.launch.py

```diff
@@ -7,17 +7,26 @@
 """
 
 import os
+import sys
 
 from ament_index_python import get_package_share_directory
 from launch import LaunchDescription
-from launch.actions import IncludeLaunchDescription
+from launch.actions import DeclareLaunchArgument, IncludeLaunchDescription
 from launch.launch_description_sources import PythonLaunchDescriptionSource
+from launch.substitutions import LaunchConfiguration
 from launch_ros.actions import Node
 
 
 def generate_launch_description():
     """This is a launch file that runs the bare minimum requirements fly a multirotor in gazebo"""
 
+    use_vimfly = False
```

Comment:
For CLI arguments inside of launch files, we should really use LaunchConfiguration and DeclareLaunchArgument. The problem with non-ROS CLI arguments is you can't set those when launching this launch file from another launch file. It would also be good to use the official ROS way of doing things as well.

Reply:
@bsutherland333 I haven't been able to get any native ROS launch tool to work. It looks like things like DeclareLaunchArgument get processed after the launch file is run, so it can't help to pass as a parameter to another file. 

The best answer I found was this...

[How to pass launch args dynamically during launch-time](https://answers.ros.org/question/376816/how-to-pass-launch-args-dynamically-during-launch-time/)

which is where the code I wrote comes from. That being said maybe there are architectural changes we could use to make this work. This same pattern is used for aircraft selection in ROSplane, so if there is a solution you have in mind please let me know!